homeassistant/components/media_player/ziggo_mediabox_xl.py

```diff
@@ -0,0 +1,174 @@
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.ziggo_mediabox_xl/
+"""
+import logging
+import socket
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    PLATFORM_SCHEMA, MediaPlayerDevice,
+    SUPPORT_TURN_ON, SUPPORT_TURN_OFF,
+    SUPPORT_NEXT_TRACK, SUPPORT_PREVIOUS_TRACK, SUPPORT_SELECT_SOURCE,
+    SUPPORT_PLAY, SUPPORT_PAUSE)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, STATE_OFF, STATE_ON, STATE_PAUSED, STATE_PLAYING)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['ziggo-mediabox-xl==1.0.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_KNOWN_DEVICES = 'ziggo_mediabox_xl_known_devices'
+
+SUPPORT_ZIGGO = SUPPORT_TURN_ON | SUPPORT_TURN_OFF | \
+    SUPPORT_NEXT_TRACK | SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK | \
+    SUPPORT_SELECT_SOURCE | SUPPORT_PLAY
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Ziggo Mediabox XL platform."""
+    from ziggo_mediabox_xl import ZiggoMediaboxXL
+
+    hass.data[DATA_KNOWN_DEVICES] = known_devices = set()
+
+    # Is this a manual configuration?
+    if config.get(CONF_HOST) is not None:
+        host = config.get(CONF_HOST)
+        name = config.get(CONF_NAME)
+    elif discovery_info is not None:
+        host = discovery_info.get('host')
+        name = discovery_info.get('name')
+    else:
+        _LOGGER.error("Cannot determine device")
+        return
+
+    # Only add a device once, so discovered devices do not override manual
+    # config.
+    hosts = []
+    ip_addr = socket.gethostbyname(host)
+    if ip_addr not in known_devices:
+        try:
+            mediabox = ZiggoMediaboxXL(ip_addr)
+            if mediabox.test_connection():
+                hosts.append(ZiggoMediaboxXLDevice(mediabox, host, name))
+                known_devices.add(ip_addr)
+            else:
+                _LOGGER.error("Can't connect to %s", host)
+        except socket.error as error:
+            _LOGGER.error("Can't connect to %s: %s", host, error)
+    else:
+        _LOGGER.info("Ignoring duplicate Ziggo Mediabox XL %s", host)
+    add_devices(hosts, True)
+
+
+class ZiggoMediaboxXLDevice(MediaPlayerDevice):
+    """Representation of a Ziggo Mediabox XL Device."""
+
+    def __init__(self, mediabox, host, name):
+        """Initialize the device."""
+        # Generate a configuration for the Samsung library
+        self._mediabox = mediabox
+        self._host = host
+        self._name = name
+        self._state = None
+
+    def update(self):
+        """Retrieve the state of the device."""
+        try:
+            if self._mediabox.turned_on():
+                if self._state != STATE_PAUSED:
+                    self._state = STATE_PLAYING
```

Comment:
This will change the state from `STATE_ON` to `STATE_PLAYING` after a call to `turn_on`. Is that anticipated?